lib/plugins/aws/package/compile/events/stream.js

```diff
@@ -273,6 +279,14 @@ class AwsCompileStreamEvents {
               };
             }
 
+            if (event.stream.patterns != null) {
+              streamResource.Properties.FilterCriteria = {
+                Filters: event.stream.patterns.map((pattern) => ({
+                  Pattern: JSON.stringify(pattern),
```

Comment:
Should we enforce stringification at this point? Do you know if we can pass CF functions safely to patterns? If yes, the stringification will break them

Reply:
Unfortunately, you cannot pass JSON in `Pattern` field. Doing so results in the following error :

```bash
An error occurred: ComputeEventSourceMappingDynamodbMyTable - Resource handler returned message: "Model validation failed (#/BatchSize: expected type: Number, found: String
  #/Enabled: expected type: Boolean, found: String
  #/FilterCriteria/Filters/0/Pattern: expected type: String, found: JSONObject)" (RequestToken: 0e5f400a-1b0b-be45-747e-5af6107a977a, HandlerErrorCode: InvalidRequest).
```

One way to cope with this would be to develop a custom function automatically matching intrinsic functions within provided pattern and wrapping the stringiffied JSON with a `Fn::Sub` use to allow for those functions. Are you sure however we should include that in the base version of this feature ? I can add in the documentation that CF intrinsic functions are not supported as is. WDYT ?

Reply:
Oh, that's too bad, but if that's the limit of CF I think we should just stick to a simple approach with stringification for now.